src/ModulusRemainder.cpp

```diff
@@ -84,7 +84,28 @@ bool reduce_expr_modulo(Expr expr, int modulus, int *remainder) {
      */
 
     if (result.modulus % modulus == 0) {
-        *remainder = result.remainder % modulus;
+        if (remainder) {
```

Comment:
Why the if statement? In what circumstances would you pass a null remainder pointer?